pkg/virt-handler/device-manager/generic_device.go

```diff
@@ -246,11 +257,21 @@ func (dpi *GenericDevicePlugin) ListAndWatch(e *pluginapi.Empty, s pluginapi.Dev
 			}
 			s.Send(&pluginapi.ListAndWatchResponse{Devices: dpi.devs})
 		case <-dpi.stop:
-			return nil
+			done = true
 		case <-dpi.done:
-			return nil
+			done = true
+		}
+		if done {
+			break
 		}
 	}
+	// Send empty list to increase the chance that the kubelet acts fast on stopped device plugins
+	// There exists no explicit way to deregister devices
+	if err := s.Send(&pluginapi.ListAndWatchResponse{Devices: emptyList}); err != nil {
+		log.DefaultLogger().Infof("%s device plugin successfully deregistered", dpi.deviceName)
```

Comment:
I am a little bit confused here. Should this say something like "Error occurred while trying to deregister"? And use this message otherwise?

Reply:
Fixed. :+1: